test/cli_wait_test.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/alibaba/pouch/test/command"
+	"github.com/alibaba/pouch/test/environment"
+
+	"github.com/go-check/check"
+	"github.com/gotestyourself/gotestyourself/icmd"
+)
+
+// PouchWaitSuite is the test suite for wait CLI.
+type PouchWaitSuite struct{}
+
+func init() {
+	check.Suite(&PouchWaitSuite{})
+}
+
+// SetUpSuite does common setup in the beginning of each test suite.
+func (suite *PouchWaitSuite) SetUpSuite(c *check.C) {
+	SkipIfFalse(c, environment.IsLinux)
+
+	environment.PruneAllContainers(apiClient)
+
+	PullImage(c, busyboxImage)
+}
+
+// TearDownTest does cleanup work in the end of each test.
+func (suite *PouchWaitSuite) TearDownTest(c *check.C) {
+}
+
+// TestWaitNonBlockedExitZero is to verify the correctness of waiting a non-blocking container with 0 exit code
+func (suite *PouchWaitSuite) TestWaitNonBlockedExitZero(c *check.C) {
+	name := "TestWaitNonBlockedExitZero"
+	command.PouchRun("run", "-d", "--name", name, busyboxImage, "sh", "-c", "true").Assert(c, icmd.Success)
+	defer DelContainerForceMultyTime(c, name)
+
+	res := command.PouchRun("wait", name)
+	res.Assert(c, icmd.Success)
+	output := res.Stdout()
+	c.Assert(output, check.Equals, fmt.Sprintf("%s\n", "0"))
+}
+
+// TestWaitBlockedExitZero is to verify the correctness of waiting a blocking container with 0 exit code
+func (suite *PouchWaitSuite) TestWaitBlockedExitZero(c *check.C) {
+	name := "TestWaitBlockedExitZero"
+	command.PouchRun("run", "-d", "--name", name, busyboxImage, "/bin/sh", "-c", "trap 'exit 0' TERM; "+
+		"while true; do usleep 10; done").Assert(c, icmd.Success)
+	defer DelContainerForceMultyTime(c, name)
+
+	chWait := make(chan string)
```

Comment:
tip: please try empty struct for the wait channel

Reply:
This channel type is `string`, can this be converted to struct{}? And this string channel's data will be transferred to variable `status` in the next.